app/blueprints/api/v4/run_history_blueprint.rb

```diff
@@ -1,3 +1,6 @@
 class Api::V4::RunHistoryBlueprint < Blueprinter::Base
   fields :id, :attempt_number, :realtime_duration_ms, :gametime_duration_ms
+
+  field(:started_at_ms) { |history, _| (history.started_at || 0).to_f * 1000 }
```

Comment:
Do these procs take 2 or 3 arguments now?  I remember there being a depreciation notice from blueprinter about something related to this, but can't remember if it was just the `if:` procs.

Reply:
According to the [CHANGELOG](https://github.com/procore/blueprinter/blob/master/CHANGELOG.md#0180----20190529), that's only for `if/unless`. Thanks, though. I wouldn't have thought to have checked.

Reply:
> Also shouldn't these be DateTime's?

Yeah unless I'm missing something, it seems like `RunHistory#started_at`/`RunHistory#ended_at` should be rendered in the API more similar to how `created_at` and `updated_at` fields are.
```ruby
> RunHistory.last.started_at
=> Thu, 26 Apr 2018 22:34:12 UTC +00:00
> Segment.last.end_ms(Run::REAL)
=> 3185758
```
It's a little confusing because `Segment#start_ms`/`Segment#end_ms` do use milliseconds to represent the elapsed time _in that run_ at the moment the segment started/ended, but for `RunHistory` we use world time for run starts and ends (and in the API we always render world times as ISO 8601 timestamps).

Reply:
Yup, `DateTime` makes way more sense. The only thing I wasn't sure about is if it's possible to have NULL values in the DB for those fields. I set it as such in the API documentation and added appropriate checks in the JS, but it might not be necessary.